dali/operators/segmentation/random_object_bbox.cc

```diff
@@ -436,12 +459,31 @@ bool RandomObjectBBox::PickForegroundBox(
   InitClassInfo(context.sample_idx);
   context.class_label = class_info_.background;
   auto &tp = *context.thread_pool;
+
+  CacheEntry *cache_entry = nullptr;
+  kernels::fast_hash_t hash = {};
+  if (use_cache_) {
+    fast_hash(hash, input.data, input.num_elements() * sizeof(T));
+    cache_entry = &cache_[hash];
+  }
+
   if (ignore_class_) {
-    int nblobs = LabelConnectedRegions(context.blobs, input, tp, -1, class_info_.background);
-    return PickBlob(context, nblobs);
+    if (!cache_entry || !cache_entry->Get(context.box_data, class_info_.background)) {
```

Comment:
```suggestion
    if (cache_entry && !cache_entry->Get(context.box_data, class_info_.background)) {
```
If `!cache_entry` then we doesn't need to call `LabelConnectedRegions` as we won't put the result in cache anyway.

Reply:
Open question is if we should waste time here to call `LabelConnectedRegions` just to put the result in cache and use it sometime later (maybe) as it won't be used now.

Reply:
Ohh, you pass the result silently through the context.
NVM